src/Analyzer.Core.UnitTests/TemplateAnalyzerTests.cs

```diff
@@ -54,6 +58,53 @@ public void AnalyzeTemplate_ValidInputValues_ReturnCorrectEvaluations(string res
             Assert.AreEqual(expectedEvaluationPassCount, evaluationsWithResults.Count(e => e.Passed));
         }
 
+        [DataTestMethod]
+        [DataRow("SimpleNestedFail.json", new int[] { 36, 43, 46, 52, 53, 54 }, DisplayName = "Simple nested template example")]
+        [DataRow("DoubleNestedFail.json", new int[] { 31, 37, 59, 53, 60,  61}, DisplayName = "Nested templates to two levels")]
+        [DataRow("InnerOuterScopeFail.json", new int[] { 49, 55, 56, 101, 107, 108, 109 }, DisplayName = "Nested template with inner and outer scope")]
+        [DataRow("ParameterPassingFail.json", new int[] { 53, 59, 62, 68, 69 }, DisplayName = "Nested template with parameters passed from parent")]
```

Comment:
Can you add a case where the nested template defines a parameter with the same name as the parent template? And validate in some way that in the context of the nested we use the proper parameter value, maybe making a rule fail that wouldn't fail with the value in the context of the parent

Reply:
I do that in the innerOuterScopeFail.json test where ftpsState is defined both in the parent and children templates. That is why line 109 fails since it is inner scope and uses the inner-defined value of ftpsState as "undesiredValue" while line 57 has no errors since it is outer scope and uses the inherited value of ftpsState as "FtpsOnly". 
It might not be sufficient in which case let me know if I should add another example.

Reply:
That's great, can you mention somewhere that the test also covers this? Maybe in the DisplayName of the test, a comment in the code, or a comment in the template?